### PR TITLE
A Single pulse gain calibration service to solve issue #197

### PR DESCRIPTION
**Pull request by relleums**
I am afraid that the only (almost clean) way to have the gain for more than one processor is the introduction of a Single pulse gain calibration service. This service can now provide several processors. This is needed now, because now there are two processors needing the gain: ```BasicExtraction``` and ```SinglePulseExtraction```

### CODE REVIEW

File: src/main/java/fact/features/singlePulse/timeSeriesExtraction/TemplatePulse.java [hunk 1]
```diff
@@ -65,4 +65,70 @@
...
+
+        final double samplingPeriodInNs = 0.5;
+        return factSinglePePulse(lengthInSlices, samplingPeriodInNs);
+
+    }
+
+    public static double factSinglePePulseIntegral() {
+
+        final double half_height = 0.5; // this is fix forever :-D
+        final double samplingPeriodInNs = 1e-3;
```
Comment (MaxNoe) [inline src/main/java/fact/features/singlePulse/timeSeriesExtraction/TemplatePulse.java]: We have 2 GSample/s, right?

File: src/main/java/fact/calibrationservice/SinglePulseGainCalibService.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import stream.annotations.Parameter;
+import stream.io.SourceURL;
+import stream.io.CsvStream;
+import stream.Data;
+import fact.Constants;
+
+/**
+ *
+ **/
+public class SinglePulseGainCalibService implements CalibrationService {
```
Comment (MaxNoe) [inline src/main/java/fact/calibrationservice/SinglePulseGainCalibService.java]: indentatio is off in this file

File: src/main/java/fact/extraction/BasicExtraction.java [hunk 3]
```diff
@@ -31,9 +30,8 @@
 	protected String outputKeyMaxAmplPos = null;
 	@Parameter(required = true, description="outputKey for the calculated photoncharge")
 	protected String outputKeyPhotonCharge = null;
-	
-	@Parameter(required = false, description = "The url to the inputfiles for the gain calibration constants",defaultValue="file:src/main/resources/defaultIntegralGains.csv")
-	protected SourceURL url = null;
+    @Parameter(required = true, description = "The calibration service for the integral single pulse gain")
```
Comment (MaxNoe) [inline src/main/java/fact/extraction/BasicExtraction.java]: This should be `@Service`, see #180

File: src/main/java/fact/extraction/SinglePulseExtraction.java [hunk 4]
```diff
@@ -80,14 +85,39 @@
     )
     protected int extractionWindowLengthInSlices = 225;
 
+    @Parameter(
```
Comment (MaxNoe) [inline src/main/java/fact/extraction/SinglePulseExtraction.java]: This should be `@Service`, see #180

**Comment by dneise** [comment]
Okay, I was working on this a bit, but during integration of the processor only I realized that calibrating the amplitudes to units of 'SinglePhotonAmplitudes' (instead of mV) does not play well with something like the saturation processor. 

Maybe however it makes sense to let the ADC saturation processor operate in the realm of voltages and only after this processor calibrate the time lines to 'SinglePhotonAmplitudes'.

The fix pixel delays should be treated much earlier I think. Within the resampling of the timelines sounds reasonable to me.

Until we didn't discuss this, I will pause working on this PR.

**Pull request merged**